src/main/java/org/jenkinsci/test/acceptance/po/PluginManager.java

```diff
@@ -213,35 +214,32 @@ public boolean installPlugins(final PluginSpec... specs) throws UnableToResolveD
                 }
             }
         } else {
-            visit("available");
+            List<PluginSpec> allRequiredPlugins = Arrays.asList(specs);
+            Map<Integer, List<String>> optionalDependencies = ucmd.get().optionalDependenciesOf(jenkins, allRequiredPlugins);
```

Comment:
:bug:

For the `uploadPlugins = false` case you can have issues using the provided update center metadata, as it may not be the same the instance is using, it would be the one to use if `uploadPlugins = true`.